sys/linux/dev_dsp.txt

```diff
@@ -54,6 +58,42 @@ ioctl$SOUND_MIXER_READ_RECSRC(fd fd_mixer, cmd const[SOUND_MIXER_READ_RECSRC], a
 ioctl$SOUND_MIXER_WRITE_VOLUME(fd fd_mixer, cmd int32[SOUND_MIXER_WRITE_VOLUME:SOUND_MIXER_WRITE_ONOFF_MAX], arg ptr[in, int32[0:100]])
 ioctl$SOUND_MIXER_READ_VOLUME(fd fd_mixer, cmd int32[SOUND_MIXER_READ_VOLUME:SOUND_MIXER_READ_ONOFF_MAX], arg ptr[out, int32])
 
+# /proc/asound/card*/oss_mixer
+resource fd_proc_mixer[fd]
+oss_mixer_proc_files = "/proc/asound/card0/oss_mixer", "/proc/asound/card1/oss_mixer", "/proc/asound/card2/oss_mixer"
+
+openat$proc_mixer(fd const[AT_FDCWD], file ptr[in, string[oss_mixer_proc_files]], flags const[O_RDWR], mode const[0]) fd_proc_mixer
+close$proc_mixer(fd fd_proc_mixer)
+read$proc_mixer(fd fd_proc_mixer, data ptr[out, array[int8]], len bytesize[data])
+write$proc_mixer(fd fd_proc_mixer, data ptr[in, array[oss_mixer_input_line]], len bytesize[data])
+
+# The input to write$proc_mixer can consist of several lines.
+# The format of each line should be:
+# <oss_mixer_name>[ '<alsa_source_names>' <index>]\n
+# e.g. "VOLUME\nBASS 'Line Capture Switch' 123\n"
+
+# 10 = '\n'
+oss_mixer_input_line {
+	name	stringnoz[oss_mixer_names]
+	body	oss_mixer_input_body
+	newline	const[10, int8]
+} [packed]
+
+oss_mixer_input_body [
+	val	oss_mixer_input_body_inner
+	void	void
+] [varlen]
+
+oss_mixer_input_body_inner {
+	sep1		stringnoz[" '"]
+	alsa_source	stringnoz[alsa_source_names]
+	sep2		stringnoz["' "]
+	index		fmt[dec, intptr[0:0]]
```

Comment:
`index` accept any positive integer less than `0x4000`. However, in qemu, only `index = 0` will work.
Similarly, the full list of `alsa_source_names` are [here](https://www.kernel.org/doc/html/v4.10/sound/designs/control-names.html#source). I only put the devices I found on qemu.
What's the best way to do this? If I change it to `intptr[0:0x4000]` I assume it will be very hard for the fuzzer to figure out only `0` is interesting.

Reply:
If it's only 0, then I think the current version is fine.